src/TypeSpec.Extension/Emitter.Csharp/src/lib/clientModelBuilder.ts

```diff
@@ -80,7 +80,7 @@ export function createModel(
 export function createModelForService(
     context: EmitContext<NetEmitterOptions>,
     service: Service
-): CodeModel {
+): [CodeModel, boolean?] {
     const emitterOptions = resolveOptions(context);
     const program = context.program;
     const sdkContext = createSdkContext(
```

Comment:
sounds good

Reply:
did not change the `createModel` signature, construct `sdkContext` in emitter instead, because `context` is needed anyway.
We do have some duplicates of constructing, such as options, will not resole them in this PR.

Reply:
updated